src/event/value.rs

```diff
@@ -89,9 +89,12 @@ impl From<BTreeMap<String, Value>> for Value {
     }
 }
 
-impl From<Vec<Value>> for Value {
-    fn from(value: Vec<Value>) -> Self {
-        Value::Array(value)
+impl<T> From<Vec<T>> for Value
```

Comment:
I'm curious to get feedback on this change. I added it to make it easier to create a `Value` from a `Vec<String>`, but it ended up having the side-effect of also implementing it for `Vec<Value>` and, I think?, will result in extra allocations for that specific case where `into()` really isn't needed.

Reply:
https://github.com/timberio/vector/pull/4093#discussion_r494991793

😄

Reply:
Aha! :smile: Thanks for digging into the assembly. I'm curious what @bruceg thinks of it.

Reply:
This actually got merged in in another PR (#4066).